packages/cadl-ranch-specs/http/azure/resource-manager/models/resources/main.tsp

```diff
@@ -38,3 +38,32 @@ union ProvisioningState {
   Deleting: "Deleting",
   Accepted: "Accepted",
 }
+
+@armResourceOperations
+interface AvailableOperations {
```

Comment:
what's an available operation?

Reply:
It's for listing available REST APIs for an RP:
doc: https://azure.github.io/typespec-azure/docs/getstarted/azure-resource-manager/step01#the-operations-interface
example: [Operation list for Microsoft.Support](https://learn.microsoft.com/en-us/rest/api/support/operations/list?view=rest-support-2024-04-01&tabs=HTTP)

Reply:
Is it possible to use this (with `@scenarioDoc` over the interface)?

```
interface Operations extends Azure.ResourceManager.Operations {}
```

ref
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/vmware/Microsoft.AVS.Management/routes.tsp#L15

Reply:
Yes, I had the same thought. 

In practice, we can. If we add `@scenario` together with `@scenarioDoc`, we could have a scenario at interface level. 
One thing(if not an issue at all) is that in cadl-ranch report, we'll have the scenario as `Operations`(or `AvailableOperations`) instead of `Operations.list`, e.g.
<img width="233" alt="Screenshot 2024-08-29 at 10 38 58" src="https://github.com/user-attachments/assets/077588f7-bf2c-4270-8eea-a5dca7b79200">
`Resources.NestedProxyResources` vs `Resources.NestedProxyResources.createOrReplace`

Though agree that we should have the test reflecting real case usage. 
We may do
```
@scenario("ListAvailableOperations")
@scenarioDoc("xxx")
interface Operations extends Azure.ResourceManager.Operations {}
```

What do you think?

Reply:
@XiaofeiCao @weidongxu-microsoft 
Does the fix in this https://github.com/Azure/cadl-ranch/pull/713/commits/dde9d5d3797af8d2ad57ffd604a7e8ec3e1d8ea4 match the ideas in the comments above?

Reply:
@v-hongli1 Exactly, thanks.